k8s-operator/apis/v1alpha1/types_proxyclass.go

```diff
@@ -209,15 +209,26 @@ type Container struct {
 	// https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context
 	// +optional
 	SecurityContext *corev1.SecurityContext `json:"securityContext,omitempty"`
-	// Container debug configuration.
-	// Options for enabling extra debug information.
-	Debug Debug `json:"debug,omitempty"`
+	// Configuration for enabling extra debug information in the container.
+	// Not recommended for production use.
+	// +optional
+	Debug *Debug `json:"debug,omitempty"`
 }
 
 type Debug struct {
-	// Endpoints enable http pprof and internal tailscaled debug metrics.
-	// Available on <pod-ip>:9002/debug/pprof and <pod-ip>:9002/debug/metrics.
-	Endpoints bool `json:"endpoints"`
+	// Enable tailscaled's HTTP pprof endpoints at <pod-ip>:9002/debug/pprof/
+	// and internal debug metrics endpoint at <pod-ip>:9002/debug/metrics, where
+	// 9002 is a container port named "debug". The endpoints and their responses
+	// may change in backwards incompatible ways in the future, and should not
+	// be considered stable.
+	//
+	// In 1.78.x and 1.80.x, this setting will default to the value of
+	// .spec.metrics.enable, and requests to the "metrics" port matching the
+	// mux pattern /debug/ will be forwarded to the "debug" port. In 1.82.x,
+	// this setting will default to false, and no requests will be proxied.
+	//
+	// +optional
+	Enable bool `json:"endpoints"`
```

Comment:
I think we wil have to make this a pointer to bool. Otherwise it is not possible to distinguish between the field explicitly set to false vs defaulted i.e for
```
apiVersion: tailscale.com/v1alpha1
kind: ProxyClass
spec:
  metrics:
    enable: true
  statefulSet:
    pod:
      tailscaleContainer:
        debug:
          enable: false
```
I would expect the debug endpoint to be off, but it's not

Reply:
The debug endpoint is indeed off for me when I apply that `ProxyClass`. Is this comment maybe based on an old commit? In the code preview I see the `Enable` field is incorrectly serialised but I fixed that in 59e1fb4.

The way I currently distinguish between explicitly off and default is to make the `Debug` field nil-able.

Reply:
I didn't spot that part of code 👍🏼